src/layout/msWorkspace/msWindow.ts

```diff
@@ -189,6 +199,25 @@ export class MsWindow extends Clutter.Actor {
 
         this.lifecycleState = lifecycleState;
         this.app = app;
+        this.createdAt = createdAt;
+        this.trackAppChanges();
+        this.connect('app-changed', () => {
```

Comment:
Just make a regular `private onAppChanged` function. We don't need slow and untyped gjs signals when this is all for internal use.